megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -3835,15 +3835,12 @@ private static ToHitData compileAeroAttackerToHitMods(Game game, Entity ae, Targ
             if (ae.usesWeaponBays() && wtype != null && weapon != null) {
 
                 // any heavy lasers
-                if (wtype.getAtClass() == WeaponType.CLASS_LASER) {
-                    for (WeaponMounted bweap : weapon.getBayWeapons()) {
-                        WeaponType bwtype = bweap.getType();
-                        if ((bwtype.getInternalName().contains("Heavy"))
-                                && (bwtype.getInternalName().contains("Laser"))) {
-                            toHit.addModifier(+1, Messages.getString("WeaponAttackAction.HeavyLaserInBay"));
-                            break;
-                        }
-                    }
+                if (wtype.getAtClass() == WeaponType.CLASS_LASER &&
+                        weapon.getBayWeapons().stream()
+                        .map(WeaponMounted::getType)
+                        .map(WeaponType::getInternalName)
+                        .anyMatch(i -> i.startsWith("CLHeavyLaser"))) {
```

Comment:
Is "CLHeavyLaser" the "Improved Heavy Laser"?

Reply:
No, it’s the unimproved versions. If that matches, it applies the penalty.